src/plone/restapi/serializer/nextprev.py

```diff
@@ -26,12 +26,15 @@ class NextPrevious(object):
 
     def __init__(self, context):
         self.context = context
-        parent = aq_parent(aq_inner(context))
-        self.nextprev = NextPreviousFixed(parent)
+        self.parent = aq_parent(aq_inner(context))
+        self.nextprev = NextPreviousFixed(self.parent)
 
     @property
     def next(self):
         """ return info about the next item in the container """
+        if getattr(self.parent, "_ordering", '') == 'unordered':
```

Comment:
@rodfersou wouldn't it make more sense to check for folders that are ordered and in that case return next/prev? With the current approach, we have to cover every single use case that we do not want to cover. I think it would be simpler and more intuitive If we just say: if this folder supports ordering, return next/prev, otherwise do nothing.

Reply:
OOTB there are three cases:
1) default ordering ( == "")
2) partial ordering (== "partial")
3) unordered (== "unordered")

And this code gives error just in the third option because when ask the position return None instead of a number.

you are suggesting to check if not "unordered" than return next / prev?

IMHO the code is written the most readable way possible, but I'm open to suggestions 🙂.